docs/src/architecture/assets.md

```diff
@@ -0,0 +1,57 @@
+
+### Fungible assets
+A fungible asset consists of a `faucet_id` of the faucet which issued the asset as well as the asset amount. The asset amount is guaranteed to be `2^63 - 1` or smaller.
+
+The most significant element of a fungible asset is set to the ID of the faucet which issued
+the asset. This guarantees the properties described above (the first bit is `ONE`).
+
+The least significant element is set to the amount of the asset. This amount cannot be greater
+than `2^63 - 1` and thus requires 63-bits to store. Elements 1 and 2 are set to `ZERO`.
+
+Note: Miden uses little-endian, so a fungible asset looks like `[amount, ZERO, e2, faucet_id]` wheras `e3` as bit ends with `1`
+
+It is impossible to find a collision between two fungible assets issued by different faucets as
+the `faucet_id` is included in the description of the asset and this is guaranteed to be different
+for each faucet as per the faucet creation logic.
+
+### Non-fungible assets
+A non-fungible asset consists of 4 field elements which are computed by hashing asset data (which can be of arbitrary length) to produce: `[e0, e1, e2, e3]`.  We then replace `e1` with the `faucet_id` that issued the asset: `[e0, faucet_id, e2, e3]` to comply with the above rule. We then set the most significant bit to `ZERO`.
+
+Note: Miden uses little-endian, so a non-fungible asset looks like `[e0, faucet_id, e2, e3]` wheras `e3` as bit ends with `0`
+
+It is impossible to find a collision between two non-fungible assets issued by different faucets as the `faucet_id` is included in the description of the non-fungible asset and this is guaranteed to be different as per the faucet creation logic. Collision resistance for non-fungible assets issued by the same faucet is `~2^95`.
+
+## Storage
+Assets are stored in accounts and notes. Both have vaults to record the `asset_id` and the respective `amount`. Accounts can store an unlimited number of assets in a tiered sparse Merkle tree called `asset_tree`. Notes can always only store one type of asset. **ToDO: Can one note store multiple NFTs?**
+
+Depending on the storage mode of the account or the note the information thereof can be private. This is true for any asset in Miden. 
+
+## Benefits
+Compared to Ethereum, assets are not tracked using global hashmaps, e.g., ERC20 contracts. Assets in Miden are stored directly in the account, like Ether in Ethereum. That provides privacy and the ability for client-side proofs, because ownership changes always only involve one account. Thus, they can happen in parallel. Additionaly, asset exchange is censorship resistant at this level, because there is no global contract the transfer must pass through. 
```

Comment:
I not have a separate "Benefits" section but would rather move it up and integrate into the intro section.

Reply:
good idea